// Copyright (c) 2020 The Open-Transactions developers
// This Source Code Form is subject to the terms of the Mozilla Public
// License, v. 2.0. If a copy of the MPL was not distributed with this
// file, You can obtain one at http://mozilla.org/MPL/2.0/.

syntax = "proto2";

package opentxs.proto;
option java_package = "org.opentransactions.proto";
option java_outer_classname = "OTBlockchainTransactionInput";
option optimize_for = LITE_RUNTIME;

import public "BlockchainInputWitness.proto";
import public "BlockchainPreviousOutput.proto";
import public "BlockchainTransactionOutput.proto";
import public "BlockchainWalletKey.proto";

message BlockchainTransactionInput {
    optional uint32 version = 1;
    optional uint32 index = 2;
    optional bytes script = 3;
    optional uint32 sequence = 4;
    optional BlockchainPreviousOutput previous = 5;
    optional BlockchainInputWitness witness = 6;
    repeated BlockchainWalletKey key = 7;
    repeated uint64 pubkey_hash = 8;
    optional uint64 script_hash = 9;
    optional bool indexed = 10;
    optional BlockchainTransactionOutput spends = 11;
}
